core/cap-0047.md

```diff
@@ -311,36 +282,31 @@ is not tied to it in any way. This allows for contract management and
 authorization to be handled in the contract using whichever custom mechanism the
 contract creator chooses.
 
-### Contracts can be updated and deleted
-The contract code is stored in a `ContractDataEntry`, so contracts can update or
-delete that entry. If the contract code is updated or deleted, the host function
-should terminate contract invocation to prevent the contract from doing
-something potentialy dangerous like writing new data after deleting a contract.
-
-TODO: Contract ID's don't have a seqnum or a nonce, so they can be recreated.
-This could be an issue if the deleted contract still has data on the ledger.
+### Contracts cannot be updated and deleted
+The contract code is stored in a `ContractDataEntry`, but contract code cannot
+be updated or deleted in the initial version. The host functions in
+[CAP-0053](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0053.md#host-function-additions)
+to update or delete `ContractDataEntry` should trap if they are used on contract
+code.
 
 ### Malicious contracts
 The validators do not have a mechanism to ban specific contracts. Any kind of
 targeted banning mechanism can be worked around quite easily by creating new
 accounts and contracts.
 
-### WASM contract code size is configurable
+### Maximum WASM contract code size is configurable
 The maximum contract size will be set during the protocol upgrade, and can be
-updated by the validators. This makes it easier to increase the maximum contract
-size without updating the XDR. Note that we will still have a hardcoded upper
-bound set in the XDR. This mechanism can also be used to reduce the contract
-size if the validators ever see a need to (TODO: size reduction is still up for
-debate).
+updated by the validators. This limit should be enforced in the
+`create_contract` host function.
 
 ### ContractID is calculated differently when created in another contract vs a classic Stellar transaction.
 Pulling contractIDs from `LedgerHeader.idPool` would be easier but it would make
 parallelizing contract creation more difficult in the future. It's also more
 difficult to determine what the contractID will be since the id pool would be
 used by offers and other contracts. This CAP uses a `Hash` instead as the
-contractID. `CreateContractTransaction` uses the source account and a salt,
-while contracts created within other contracts use the parent contractID and a
-salt.
+contractID. Creating a contract using `InvokeHostFunctionOp` uses the source
```

Comment:
Why does this work differently from the actual host function? It's supposed to be actually calling the host function, so I would expect it to work the same.

Reply:
This is a remnant of the old method that I forgot to remove. I'll update this section.